components/automate-ui/src/app/pages/api-token/details/api-token-details.component.ts

```diff
@@ -56,20 +56,33 @@ export class ApiTokenDetailsComponent implements OnInit, OnDestroy {
       status: [initialStatus],
       projects: [[]]
     });
+  }
+
+  ngOnInit(): void {
     this.store.pipe(
       select(isIAMv2),
       takeUntil(this.isDestroyed))
       .subscribe(latest => {
         this.isIAMv2 = latest;
       });
-  }
 
-  ngOnInit(): void {
     this.store.pipe(
-      select(apiTokenFromRoute),
+      select(routeParams),
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.store.dispatch(new GetToken({ id }));
+      });
+
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(apiTokenFromRoute)
+    ]).pipe(
+      filter(([gStatus, _]) => gStatus === EntityStatus.loadingSuccess),
       filter(identity),
```

Comment:
This isn't gonna quite be what you want now. `filter(identity)` is shorthand for:

`filter(x => return x)`

Which in this case is:

```
filter([gStatus, apiTokenFromRoute] => return [gStatus, apiTokenFromRoute])
```

That could create a race condition where the `getStatus` has been defined but `apiTokenFromRoute` is still `undefined` (just ran into that on my end!) and `filter(identify)` will return true there.

You can just do:

```
import { isNil } from 'lodash/fp';
filter(([_, tokenState]) => !isNil(tokenState))
```

Reply:
good find!